src/nnet3/nnet-simple-component.cc

```diff
@@ -2635,6 +2795,19 @@ int32 ConvolutionComponent::NumParameters() const {
   return (filter_params_.NumCols() + 1) * filter_params_.NumRows();
 }
 
+void ConvolutionComponent::Vectorize(VectorBase<BaseFloat> *params) const {
```

Comment:
The only situations where it makes sense to compare these vectors is where
the nnets have the same structure, so no, don't include those things in the
params.  They are supposed to be just the parameters, and in some
situations we might scale them or average them, which doesn't make sense
for integer values.
Dan


On Wed, Nov 4, 2015 at 4:26 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet3/nnet-simple-component.cc
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43941818>:
>
> > @@ -2635,6 +2795,19 @@ int32 ConvolutionComponent::NumParameters() const {
> >    return (filter_params_.NumCols() + 1) * filter_params_.NumRows();
> >  }
> >
> > +void ConvolutionComponent::Vectorize(VectorBase<BaseFloat> *params) const {
>
> I don't exactly know where Vectorize and Unvectorize are going to be used.
> But isn't it important to even add the hyper-parameters like filter and
> input dimensions to the params ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r43941818>.
>